application/controllers/admin/Labels.php

```diff
@@ -740,4 +741,64 @@ protected function renderWrappedTemplate($sAction = 'labels', $aViewUrls = array
 
         parent::renderWrappedTemplate($sAction, $aViewUrls, $aData, $sRenderFile);
     }
+
+    /**
+     * Outputs json with the html for a new label row, for each language.
+     * Eg: {"en":"<tr>...</tr>", "es":"<tr>...</tr>"}
+     * Called by ajax from the label set editor.
+     * @param int $lid
+     * @param string $newId
+     * @param string $code
+     * @param int $assessmentValue
+     * @param string $title
+     */
+    public function getLabelRowForAllLanguages($lid, $newId, $code, $assessmentValue = 0, $title = '')
+    {
+        if (!Permission::model()->hasGlobalPermission('labelsets', 'update')) {
```

Comment:
Read permission should be enough, if there's no write in the method?

Reply:
This is because this function is called in the process of editing labels.

Reply:
You shouldn't lock a function to its context, tho, you might prevent reuse that way.

Reply:
But this is an action in the controller. Not expecting to reuse this.

Reply:
> But this is an action in the controller. Not expecting to reuse this.


You can get the json just to show it to user . It's a public function. 
Why disallow show lable if user have read rights ?

This public function didn't update label : then need only read.

Reply:
You're not expecting to reuse it _now_, but who knows what will happen 2-5 years from now.

Reply:
Do you expect to reuse controller actions?
Anyway, already changed the permission to read :)

Reply:
"getLabelRowForAllLanguages" sounds like something that could be used in, say, question editor. Na ja. Thanks!